### PR TITLE
fix(panel): Conditionally set tabindex on scrollable area. 

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #1642

## Summary

fix(panel): Conditionally set tabindex on scrollable area. (#1642)

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco for some reason, this is throwing errors in the `it("has slots", () => slots("calcite-panel", SLOTS));` test.

I think the resizeObserver is causing some issues here?

There's also an issue with the `it("can be disabled", () => disabled(`<calcite-panel>scrolling content</calcite-panel>`));` test as well since the tabIndex is no longer there in this scenario.

**Comment by driskull** [comment]
```
FAIL src/components/panel/panel.e2e.ts (19.543 s)
  ● calcite-panel › has slots

    JSHandle@error JSHandle@node 
    Location: http://localhost:3333/build/index-fec7d2f1.js:2933:62

      at node_modules/@stencil/core/testing/index.js:3[889](https://github.com/Esri/calcite-components/runs/6165397197?check_suite_focus=true#step:6:889):18
      at node_modules/puppeteer/vendor/mitt/src/index.ts:88:75
          at Array.map (<anonymous>)
      at Object.emit (node_modules/puppeteer/vendor/mitt/src/index.ts:88:56)
      at Page.emit (node_modules/puppeteer/src/common/EventEmitter.ts:102:18)
      at Page._addConsoleMessage (node_modules/puppeteer/src/common/Page.ts:1328:10)
      at Page._onConsoleAPI (node_modules/puppeteer/src/common/Page.ts:1254:10)
      at node_modules/puppeteer/src/common/Page.ts:558:59
      at node_modules/puppeteer/vendor/mitt/src/index.ts:88:75
          at Array.map (<anonymous>)

  ● calcite-panel › has slots

    expect(jest.fn()).not.toHaveBeenCalled()

    Expected number of calls: 0
    Received number of calls: 1

    1: "JSHandle@error JSHandle@node·
    Location: http://localhost:3333/build/index-fec7d2f1.js:2933:62"

       8 |
       9 | // eslint-disable-next-line jest/no-standalone-expect
    > 10 | afterEach(() => expect(globalError).not.toHaveBeenCalled());
         |                                         ^
      11 |
      12 | afterAll(() => {
      13 |   globalError.mockClear();

      at Object.<anonymous> (src/tests/setupTests.ts:10:41)
          at runMicrotasks (<anonymous>)
```

File: src/components/panel/panel.tsx [hunk 1]
```diff
@@ -150,6 +152,10 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
 
   panelScrollEl: HTMLElement;
 
+  resizeObserver = !Build.isTesting
```
Comment (driskull) [inline src/components/panel/panel.tsx:155]: Was causing an error in the `slot` commonTest if this was not there.

File: src/tests/commonTests.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { JSX } from "../components";
 import { toHaveNoViolations } from "jest-axe";
 import axe from "axe-core";
 import { config } from "../../stencil.config";
-import { GlobalTestProps, waitForAnimationFrame } from "./utils";
+import { GlobalTestProps } from "./utils";
```
Comment (jcfranco) [inline src/tests/commonTests.ts:6]: I can submit a PR for this to split it out.

File: src/components/panel/panel.e2e.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ describe("calcite-panel", () => {
 
   it("has slots", () => slots("calcite-panel", SLOTS));
 
-  it("can be disabled", () => disabled(`<calcite-panel>scrolling content</calcite-panel>`));
+  it("can be disabled", () => disabled(`<calcite-panel dismissible>scrolling content</calcite-panel>`));
```
[the remark this answers is not in the document]
Comment (driskull) [reply src/components/panel/panel.e2e.ts:25]: Correct. non-dismissible panels are not interactive unless there's some scrolling now.

File: src/components/panel/panel.tsx (hunk 1 above)
Comment (driskull) [reply src/components/panel/panel.tsx:155]: Correct. slot helper error will appear. I'll check it in so you can see. I'm seeing if I can figure out why.

File: src/tests/commonTests.ts (hunk 2 above)
Comment (driskull) [reply src/tests/commonTests.ts:6]: Created https://github.com/Esri/calcite-components/pull/4484

**Comment by driskull** [comment]
@jcfranco can you check this one out and see why it's failing these tests?

https://github.com/Esri/calcite-components/pull/4374#issuecomment-1109051478

If I remove the resizeObserver the failures go away.

**Pull request merged**